jade-repl.el

```diff
@@ -194,31 +195,89 @@ When ERROR is non-nil, use the error face."
     (jade-repl-insert-prompt)
     (run-hooks 'jade-repl-evaluate-hook)))
 
-(defun jade-repl-emit-console-message (string &optional level)
-  "Emit a console message STRING.
-LEVEL is a string representing the logging level, it can be
-\"log\", \"warn\", \"debug\" or \"error\"."
+(defun jade-repl-emit-console-message (message)
+  "Emit a console message.
+MESSAGE is a map ((text . ) (level . ) (type . ) (description . ) ...)."
   (with-current-buffer (jade-repl-get-buffer)
     (save-excursion
-      (let* ((error (string= level "error"))
-             (face (when error 'jade-repl-error-face))
-             (message (if level
-                          (concat level ": " string)
-                        string)))
+      (let ((level (map-elt message 'level))
+            (text (map-elt message 'text))
+            (parameters (map-elt message 'parameters))
+            (url (map-elt message 'url))
+            (line (map-elt message 'line)))
         (goto-char jade-repl-output-end-marker)
-        (insert "\n")
         (set-marker jade-repl-output-start-marker (point))
-        (insert
-         (ansi-color-apply
-          (propertize message
-                      'font-lock-face (or face 'jade-repl-stdout-face)
-                      'rear-nonsticky '(font-lock-face))))
+        (jade-repl--emit-values text
+                                (seq-map #'jade-webkit--value parameters)
```

Comment:
There cannot be a reference to `jade-webkit-*' functions here, or we're breaking all other (future) backends.
It means that the message has to be converted prior to being passed to `jade-repl', in `jade-webkit.el'.